### PR TITLE
add ctx->handling and ctx->vsl for cli

### PR DESCRIPTION
**Pull request by nigoroll**
This implements one of the possible options to approach #2997

### CODE REVIEW

File: bin/varnishd/cache/cache_vcl.c [hunk 1]
```diff
@@ -85,6 +86,8 @@ VCL_Bo2Ctx(struct vrt_ctx *ctx, struct busyobj *bo)
 	ctx->sp = bo->sp;
 	ctx->now = bo->t_prev;
 	ctx->ws = bo->ws;
+	ctx->handling = &bo->wrk->handling;
+	*ctx->handling = 0;
```
Comment (bsdphk) [inline bin/varnishd/cache/cache_vcl.c]: Isn't this already done by vcl_call_method() in cache_vrt_vcl ?

Comment (nigoroll) [reply bin/varnishd/cache/cache_vcl.c]: `vcl_call_method()` uses a local `struct vrt_ctx`, as does `VDI_GetHdr()`, so there is no effect of the former on the latter.
But you rightly pointed out that my suggestion duplicates these two lines, I will check if they could be removed from `vcl_call_method()`.

Comment (nigoroll) [reply bin/varnishd/cache/cache_vcl.c]: sorry, I was looking at master, I actually had already suggested to remove the duplication here:
https://github.com/varnishcache/varnish-cache/pull/2998/commits/218c818023dfb3446e413dd00b64c755be764876#diff-f00ee27228a92dc1471fc250affb40c9L444

**Comment by nigoroll** [comment]
@bsdphk isn't this https://github.com/varnishcache/varnish-cache/pull/2998/commits/51fb3d4fef45c0911283a932123f1024ec680157 ?

**Comment by nigoroll** [comment]
@Dridi I agree that we should continue in this direction.

**Pull request merged**